src/stratis/run.rs

```diff
@@ -90,11 +94,18 @@ pub fn run(sim: bool) -> StratisResult<()> {
     runtime.block_on(async move {
         async fn start_threads<E>(engine: Arc<E>, sim: bool) -> StratisResult<()> where E: 'static + Engine {
             let (trigger, should_exit) = channel(1);
+
+            #[cfg(any(feature = "dbus_enabled", feature = "min"))]
             let (udev_sender, udev_receiver) = unbounded_channel::<UdevEngineEvent>();
+            #[cfg(all(not(feature = "dbus_enabled"), not(feature = "min")))]
+            let (udev_sender, _) = unbounded_channel::<UdevEngineEvent>();
```

Comment:
Both of the last two approaches seems to cause problems. The last one would generate a channel that would progressively take up more memory and never be drained and this one will generate a channel disconnect error every time a udev event is received. Neither seems ideal. I would recommend either requiring an IPC mechanism for the time being or having a separate thread to trigger udev event handling in the engine even if there is no IPC mechanism available.